ivadomed/ivadomed
# Fix segment functions for models without pre-processing transforms

## Checklist

#### GitHub


#### PR contents


## Description
Fixes #873, where the command `segment` and the function `segment_volume` would not work with a model without pre-processing because `seg_pair_sclice['gt']` is `[]` instead of `None` in the loader.

I changed the if condition in `getitem` of the classes `MRI2DSegmentationDataset` and `MRI3DSubVolumeSegmentationDataset`, both `seg_pair_sclice['gt'] = []` and `seg_pair_sclice['gt'] = None` are now working with the segment functions.

I added a test for the 2D case i.e. a model without pre-processing, only with `NormalizeInstance`.

I cannot add a test for the 3D case because there are constraints on the size of `LENGTH_3D` and `STRIDE_3D` of the subvolumes [here](https://github.com/ivadomed/ivadomed/blob/7a595ae39e94085837582f7176b8ec59bd9a2bc9/ivadomed/loader/mri3d_subvolume_segmentation_dataset.py#L92) which require a `CenterCrop` to a specific size in the pre-processing. So it is not likely that the same issue would happen in 3D but I thought correcting both 2D and 3D was a good idea.

## To test the PR:
- On master branch, train a model without transformations with the following config file (~5 minutes on cpu: this is a test config with the bare minimum, 1 epoch, 1 subject, 1 contrast, etc.) [config_test_873.zip](https://github.com/ivadomed/ivadomed/files/6880921/config_test_873.zip)
- With the same config file, run the `segment` command, you should have the same error as described [here](https://github.com/ivadomed/ivadomed/issues/873#issue-951963063).
- On this branch, run the `segment` command again, the segmentation should run without error.
